cdap-security/src/test/java/co/cask/cdap/security/store/client/RemoteSecureStoreTest.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.security.store.client;
+
+import co.cask.cdap.api.security.store.SecureStoreData;
+import co.cask.cdap.api.security.store.SecureStoreMetadata;
+import co.cask.cdap.common.HttpExceptionHandler;
+import co.cask.cdap.common.SecureKeyNotFoundException;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.proto.id.SecureKeyId;
+import co.cask.http.AbstractHttpHandler;
+import co.cask.http.HttpResponder;
+import co.cask.http.NettyHttpService;
+import com.google.common.collect.ImmutableMap;
+import com.google.gson.Gson;
+import io.netty.handler.codec.http.DefaultHttpHeaders;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import org.apache.twill.discovery.Discoverable;
+import org.apache.twill.discovery.InMemoryDiscoveryService;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+/**
+ * Tests for {@link RemoteSecureStore}.
+ */
+public class RemoteSecureStoreTest {
+  private static NettyHttpService httpService;
+  private static RemoteSecureStore remoteSecureStore;
+
+  @BeforeClass
+  public static void setUp() throws Exception {
+    // Starts a mock server to handle remote secure store requests
+    httpService = NettyHttpService.builder("remoteSecureStoreTest")
+      .setHttpHandlers(new SecureStoreTestHandler())
```

Comment:
Why use a test one? Can't use the actual `SecureStoreHandler` with a file backed store store?